fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/api/v2/search/SavingsAccountTransactionsSearchApiResource.java

```diff
@@ -0,0 +1,66 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.portfolio.savings.api.v2.search;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import lombok.RequiredArgsConstructor;
+import org.apache.fineract.infrastructure.core.serialization.DefaultToApiJsonSerializer;
+import org.apache.fineract.infrastructure.core.service.Page;
+import org.apache.fineract.infrastructure.core.service.PagedRequest;
+import org.apache.fineract.portfolio.savings.api.SavingsAccountTransactionsApiResourceSwagger;
+import org.apache.fineract.portfolio.savings.data.search.SavingsAccountTransactionSearchData;
+import org.apache.fineract.portfolio.savings.domain.search.SavingsTransactionSearch;
+import org.apache.fineract.portfolio.savings.service.search.SavingsAccountTransactionsSearchServiceImpl;
+import org.springframework.stereotype.Component;
+
+@Path("/v2/savingsaccounts/{savingsId}/transactions")
+@Component
+@Tag(name = "Savings Account Transactions Search")
+@RequiredArgsConstructor
+public class SavingsAccountTransactionsSearchApiResource {
+
+    private final SavingsAccountTransactionsSearchServiceImpl service;
+    private final DefaultToApiJsonSerializer<SavingsAccountTransactionSearchData> toApiJsonSerializer;
+
+    @POST
+    @Path("search")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Search Savings Account Transactions")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = SavingsAccountTransactionsApiResourceSwagger.SavingsAccountTransactionsSearchResponse.class))) })
+    public String searchTransactions(@PathParam("savingsId") @Parameter(description = "savingsId") final Long savingsId,
```

Comment:
Please use the Spring DATA Page instead of the "own"... 
The idea is to move towards the standard solutions and let the "custom" ones got deprecated slowly.

Reply:
Hi Adam,

I had also considered utilizing the Spring Data Page instead of the legacy response. However, Peter expressed a preference for the response to resemble the legacy implementation. If it is necessary for me to accommodate the legacy response fields while incorporating the spring data Page, it will require a refactoring of all the transaction-related entities.
I am concerned that making those changes may potentially disrupt the existing implementations.

Reply:
@Mk9894 I see.. would you please change than only to be this v1?

The idea is v2 is using the new way of generating the swagger definitions and using the Spring data page and some other enhancements, it this is to be support the legacy pagination, it should remain v1.

is that fine?

Reply:
Done.